controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentTrigger.java

```diff
@@ -292,8 +292,10 @@ private ApplicationVersion targetApplication(Application application, Change cha
                             if (completedAt.isPresent())
                                 jobs.add(deploymentJob(application, target, application.change(), job, reason, completedAt.get(), stepJobs));
                         }
-                        else if (testJobs == null)
-                            testJobs = testJobsFor(application, target, "Testing deployment for " + job.jobName(), completedAt.orElse(clock.instant()));
+                        else if (testJobs == null) {
+                            if ( ! alreadyTriggered(application, target))
```

Comment:
Otherwise, an application with two upgrade targets (due to previous failed deployments, or, later, due to an aborted change) would start tests for the other target immediately after completing them for the first, disregarding the delay.